src/consensus/aft/raft.h

```diff
@@ -825,6 +836,67 @@ namespace aft
         ccf::NodeMsgType::consensus_msg, to, response);
     }
 
+    void send_append_entries_signed_response(NodeId to, ccf::Signature& sig)
+    {
+      LOG_DEBUG_FMT(
+        "Send append entries signed response from {} to {} for index {}",
+        state->my_node_id,
+        to,
+        state->last_idx);
+
+      SignedAppendEntriesResponse r = {
+        {raft_append_entries_signed_response, state->my_node_id},
+        state->current_view,
+        state->last_idx,
+        static_cast<uint32_t>(sig.sig.size()),
+        {}};
+      r.sig.fill(0);
+      std::copy(sig.sig.begin(), sig.sig.end(), r.sig.data());
+
+      auto progress_tracker = store->get_progress_tracker();
+      if (progress_tracker != nullptr)
+      {
+        progress_tracker->add_signature(
+          r.term, r.last_log_idx, r.from_node, r.signature_size, r.sig);
+      }
+
+      channels->send_authenticated(ccf::NodeMsgType::consensus_msg, to, r);
+    }
+
+    void recv_append_entries_signed_response(const uint8_t* data, size_t size)
+    {
+      SignedAppendEntriesResponse r;
+
+      try
+      {
+        r = channels->template recv_authenticated<SignedAppendEntriesResponse>(
+          data, size);
+      }
+      catch (const std::logic_error& err)
+      {
+        LOG_FAIL_FMT(err.what());
```

Comment:
Not sure what's in what() but it may be prudent to have a hardcoded FAIL message and have this only in DEBUG, like we've done elsewhere, for confidentiality reasons.